rasa_sdk/endpoint.py

```diff
@@ -39,120 +54,63 @@ def create_argument_parser():
     return parser
 
 
-def endpoint_app(cors_origins=None, action_package_name=None):
-    app = Flask(__name__)
+def create_app(
+    action_package_name: Union[Text, types.ModuleType],
+    cors_origins: Union[Text, List[Text], None] = "*",
+) -> Sanic:
+    app = Sanic(__name__, configure_logging=False)
 
-    if not cors_origins:
-        cors_origins = []
+    configure_cors(app, cors_origins)
 
     executor = ActionExecutor()
     executor.register_package(action_package_name)
 
-    CORS(app, resources={r"/*": {"origins": cors_origins}})
-
-    @app.route("/health", methods=["GET", "OPTIONS"])
-    @cross_origin(origins=cors_origins)
-    def health():
+    @app.get("/health")
+    async def health(request):
         """Ping endpoint to check if the server is running and well."""
-        return jsonify({"status": "ok"})
+        body = {"status": "ok"}
+        return response.json(body, status=200)
 
-    @app.route("/webhook", methods=["POST", "OPTIONS"])
-    @cross_origin()
-    def webhook():
+    @app.post("/webhook")
+    async def webhook(request):
         """Webhook to retrieve action calls."""
         action_call = request.json
-        check_version_compatibility(action_call.get("version"))
+        utils.check_version_compatibility(action_call.get("version"))
```

Comment:
can we please also handle the case that `action_call` is `None` (sorry, I know it's not your code, but it should be a quick fix)